examples/monoio_tlsclient.rs

```diff
@@ -0,0 +1,172 @@
+use hyper::Request;
+use hyper::Uri;
+use monoio::net::TcpStream;
+use std::future::Future;
+use std::sync::Arc;
+use tokio_rustls::rustls::ClientConfig;
+use tokio_rustls::rustls::OwnedTrustAnchor;
+use tokio_rustls::TlsConnector;
+
+#[allow(deprecated)]
+fn tls_connector() -> Result<TlsConnector> {
+  let mut root_store = tokio_rustls::rustls::RootCertStore::empty();
+
+  root_store.add_server_trust_anchors(
+    webpki_roots::TLS_SERVER_ROOTS.0.iter().map(|ta| {
+      OwnedTrustAnchor::from_subject_spki_name_constraints(
+        ta.subject,
+        ta.spki,
+        ta.name_constraints,
+      )
+    }),
+  );
+
+  let config = ClientConfig::builder()
+    .with_safe_defaults()
+    .with_root_certificates(root_store)
+    .with_no_client_auth();
+
+  Ok(TlsConnector::from(Arc::new(config)))
+}
+
+async fn handle_websocket_upgrade(
+  uri: Uri,
+  port: u16,
+) -> Result<(), WebSocketError> {
+  // 1. 创建HTTP客户端
+  let host = uri.host().expect("uri has no host");
+  let port = uri.port_u16().unwrap_or(port);
+  let addr = format!("{}:{}", host, port);
+  let stream = TcpStream::connect(&addr).await?;
+  let tcp_stream = HyperConnection(stream);
+  println!("Connected to: {:?}", addr);
+  let domain =
+    tokio_rustls::rustls::ServerName::try_from(uri.to_string().as_str())
+      .map_err(|_| {
+        std::io::Error::new(std::io::ErrorKind::InvalidInput, "invalid dnsname")
+      })?;
+
+  println!("Domin found: {:?}", domain);
+  let tls_connector = tls_connector().unwrap();
+  let tls_stream = tls_connector.connect(domain, tcp_stream).await.unwrap();
+
+  let req = Request::builder()
+    .method("GET")
+    .uri(format!("wss://{}/ws/btcusdt@bookTicker", &addr))
+    .header("Host", &addr)
+    .header("Upgrade", "websocket")
+    .header("Connection", "Upgrade")
+    .header(
+      "Sec-WebSocket-Key",
+      fastwebsockets::handshake::generate_key(),
+    )
+    .header("Sec-WebSocket-Version", "13") // WebSocket 版本
+    .body(Body::empty())
+    .expect("Failed to build request");
+
+  let (mut ws, _) =
+    fastwebsockets::handshake::client(&HyperExecutor, req, tls_stream).await?;
+  println!("WebSocket handshake succeeded");
+  loop {
+    let msg = match ws.read_frame().await {
+      Ok(msg) => msg,
+      Err(e) => {
+        println!("Error: {}", e);
+        ws.write_frame(Frame::close_raw(vec![].into())).await?;
+        break;
+      }
+    };
+
+    match msg.opcode {
+      OpCode::Text => {
+        let payload =
+          String::from_utf8(msg.payload.to_vec()).expect("Invalid UTF-8 data");
+        // Normally deserialise from json here, print just to show it works
+        println!("{:?}", payload);
+      }
+      OpCode::Close => {
+        break;
+      }
+      _ => {}
+    }
+  }
+  Ok(())
+}
+
+#[monoio::main]
+async fn main() {
+  // !!!!! do not use proxychains or other proxy tools, the tcp steam connect may be failed !!!!
+  let uri: Uri = "data-stream.binance.com".parse::<hyper::Uri>().unwrap();
```

Comment:
I think we should connect to a local echo server instead of a remote service like binance.